api/testing/restapi_test.go

```diff
@@ -0,0 +1,686 @@
+						Name: name,
+					},
+					Spec: &api.VolumeSpec{
+						Size: size,
+					},
+				},
+			}, nil).
+			Times(1),
+	)
+
+	driverclient := volumeclient.VolumeDriver(ts.client)
+
+	res, err := driverclient.Inspect([]string{id})
+
+	assert.Nil(t, err)
+	assert.NotNil(t, res)
+	assert.NotEmpty(t, res)
+	assert.Equal(t, res[0].GetId(), id)
+
+}
+
+func TestVolumeSetSuccess(t *testing.T) {
+
+	ts := setupMocks()
+
+	defer ts.Stop()
+
+	var err error
+	ts.client, err = volumeclient.NewDriverClient(host, driver, version, "")
+
+	assert.Nil(t, err)
+
+	// create a volume request
+
+	name := "myvol"
+	id := "myid"
+	size := uint64(10)
+
+	req := &api.VolumeSetRequest{
+		Options: map[string]string{},
+		Action: &api.VolumeStateAction{
+			Attach: api.VolumeActionParam_VOLUME_ACTION_PARAM_ON,
+			Mount:  api.VolumeActionParam_VOLUME_ACTION_PARAM_ON},
+		Locator: &api.VolumeLocator{Name: name},
+		Spec:    &api.VolumeSpec{Size: size},
+	}
+
+	gomock.InOrder(
+		ts.MockDriver().
+			EXPECT().
+			Set(id, req.GetLocator(), req.GetSpec()).
+			Return(nil),
+
+		ts.MockDriver().
+			EXPECT().
+			Inspect([]string{id}).
+			Return([]*api.Volume{
+				&api.Volume{
+					Id: id,
+					Locator: &api.VolumeLocator{
+						Name: name,
+					},
+					Spec: &api.VolumeSpec{
+						Size: size,
+					},
+				},
+			}, nil),
+	)
+
+	// create driver client
+
+	driverclient := volumeclient.VolumeDriver(ts.client)
+
+	res := driverclient.Set(id, req.GetLocator(), req.GetSpec())
+	assert.Nil(t, res)
+
+}
+
+func TestVolumeAttachSuccess(t *testing.T) {
+
+	ts := setupMocks()
+
+	defer ts.Stop()
+
+	var err error
+	ts.client, err = volumeclient.NewDriverClient(host, driver, version, "")
+
+	assert.Nil(t, err)
+
+	name := "myvol"
+	id := "myid"
+	size := uint64(10)
+
+	req := &api.VolumeSetRequest{
+		Options: map[string]string{},
+		Action: &api.VolumeStateAction{
+			Attach: api.VolumeActionParam_VOLUME_ACTION_PARAM_ON},
+		Locator: &api.VolumeLocator{Name: name},
+		Spec:    &api.VolumeSpec{Size: size},
+	}
+
+	gomock.InOrder(
+		ts.MockDriver().
+			EXPECT().
+			Attach(id, gomock.Any()).
+			Return("", nil),
+
+		ts.MockDriver().
+			EXPECT().
+			Inspect([]string{id}).
+			Return([]*api.Volume{
+				&api.Volume{
+					Id: id,
+					Locator: &api.VolumeLocator{
+						Name: name,
+					},
+					Spec: &api.VolumeSpec{
+						Size: size,
+					},
+				},
+			}, nil),
+	)
+
+	// create driver client
+
+	driverclient := volumeclient.VolumeDriver(ts.client)
+
+	_, err = driverclient.Attach(id, req.GetOptions())
+
+	assert.Nil(t, err)
+
+}
+
+func TestVolumeAttachFailed(t *testing.T) {
+
+	ts := setupMocks()
+
+	defer ts.Stop()
+
+	var err error
+	ts.client, err = volumeclient.NewDriverClient(host, driver, version, "")
+
+	assert.Nil(t, err)
+
+	name := "myvol"
+	id := "myid"
+	size := uint64(10)
+
+	req := &api.VolumeSetRequest{
+		Options: map[string]string{},
+		Action: &api.VolumeStateAction{
+			Attach: api.VolumeActionParam_VOLUME_ACTION_PARAM_ON},
```

Comment:
nit: 
Attach: api.VolumeActionParam_VOLUME_ACTION_PARAM_ON},
should be
Attach: api.VolumeActionParam_VOLUME_ACTION_PARAM_ON,
}

fix at other places as well.